test/ha/activator_test.go

```diff
@@ -38,19 +38,32 @@ import (
 const (
 	activatorDeploymentName = "activator"
 	activatorLabel          = "app=activator"
-	minProbes               = 400  // We want to send at least 400 requests.
-	SLO                     = 0.99 // We permit 0.01 of requests to fail due to killing the Activator.
+	minProbes               = 400 // We want to send at least 400 requests.
 )
 
+func TestActivatorHAGraceful(t *testing.T) {
+	testActivatorHA(t, true, 1)
+}
+
+func TestActivatorHANonGraceful(t *testing.T) {
+	testActivatorHA(t, false, 0.95)
+}
+
 // The Activator does not have leader election enabled.
 // The test ensures that stopping one of the activator pods doesn't affect user applications.
 // One service is probed during activator restarts and another service is used for testing
 // that we can scale from zero after activator restart.
-func TestActivatorHA(t *testing.T) {
+func testActivatorHA(t *testing.T, allowGraceful bool, slo float64) {
```

Comment:
```suggestion
func testActivatorHA(t *testing.T, gracePeriod *int64, slo float64) {
```
which also opens window to setting it to arbitrary value?

how about this? and then instead

Reply:
hm, not convinced. Setting it to `nil` to get the default period then?

Reply:
nil or ptr(300)

Reply:
Thinking about any case when we'd like to set something different then 0 or default. But I can't come up with any. We don't give any guidance for the grace period so users will just use the default, IMO. And crashes won't ask how long they should wait :)

Reply:
2 hours ago we didn't think about any value but 0 :-)

Reply:
I dunno if arbitrary values are practical in production though? The deployment has a limit set, and the default will value that. Anything in between I'd assume is roughly equal to a hard-shutdown i.e. "undefined behavior" (though of course this technically only needs to be as long as max(hardcoded 30s grace, longest request) in practice)

Reply:
also we get rid of the `if` :)